### PR TITLE
Mkaito/fix default nix

### PR DESCRIPTION
**Pull request by mkaito**
[//]: # (This is a template of a pull request template.)
[//]: # (You should modify it considering specifics of a particular repository and put it there.)
[//]: # (Comments like this are meta-comments, they shouldn't be present in the final template.)
[//]: # (Comments starting with '

[//]: # (Here you can add a link to the corresponding issue tracker, e. g. https://issues.serokell.io/issue/SDAO-)
## Related issue(s)
None

#### Related changes (conditional)

- Tests
  - [ ] If I added new functionality, I added tests covering it.
  - [ ] If I fixed a bug, I added a regression test to prevent the bug from
        silently reappearing again.

[//]: # (Add more docs here if you have them in the repository)
- Documentation
  - [ ] I checked whether I should update the docs and did so if necessary:
    - [README](../tree/master/README.md)
    - Haddock

[//]: # (Mostly for public repositories)
[//]: # (Recording changes is optional, depends on repository, useful for some libs)
- Public contracts
  - [ ] Any modifications of public contracts comply with the [Evolution
  of Public Contracts](https://www.notion.so/serokell/Evolution-of-Public-Contracts-2a3bf7971abe4806a24f63c84e7076c5) policy.
  - [ ] I added an entry to the [changelog](../tree/master/CHANGES.md) if my changes are visible to the users
        and
  - [ ] provided a migration guide for breaking changes if possible

#### Stylistic guide (mandatory)

[//]: # (Update link to style guide if necesary or remove if it's not present)

- [ ] My commits comply with [the policy used in Serokell](https://www.notion.so/serokell/Where-and-how-to-commit-your-work-58f8973a4b3142c8abbd2e6fd5b3a08e).
- [ ] My code complies with the [style guide](../tree/master/docs/code-style.md).

### CODE REVIEW

File: default.nix [hunk 1]
```diff
@@ -1,24 +1,29 @@
...
+  haskellPackages = import ./pkgs.nix { inherit haskell pkgs; };
+  package = haskellPackages.stkr-token;
 
-  hsTools = with haskellPackages; [
-    cabal-install hpack
-    hlint hdevtools
-    morley
-  ];
+  devEnv = package.env.overrideAttrs(attr: {
+    buildInputs = with haskellPackages;
```
Comment (kirelagin) [inline default.nix]: All these are `nativeBuildIntputs`.

File: default.nix [hunk 2]
```diff
@@ -1,24 +1,29 @@
...
-  drv = haskellPackages.stkr-token;
+  haskellPackages = import ./pkgs.nix { inherit haskell pkgs; };
+  package = haskellPackages.stkr-token;
 
-  hsTools = with haskellPackages; [
-    cabal-install hpack
-    hlint hdevtools
-    morley
-  ];
+  devEnv = package.env.overrideAttrs(attr: {
```
Comment (kirelagin) [inline default.nix]:
```suggestion
  devEnv = package.env.overrideAttrs (attr: {
```

File: default.nix [hunk 3]
```diff
@@ -1,24 +1,29 @@
...
+    buildInputs = with haskellPackages;
+      [
+        cabal-install hpack
+        hlint hdevtools
+        morley
```
Comment (kirelagin) [inline default.nix]: Except for maybe this one, I’m not sure.

File: default.nix (hunk 1 above)
Comment (yorickvP) [inline default.nix]:
```suggestion
    nativeBuildInputs = with haskellPackages;
```

File: default.nix (hunk 3 above)
Comment (yorickvP) [inline default.nix]:
```suggestion
        ];
        buildInputs = [
          morley
```

File: default.nix (hunk 1 above)
Comment (gpevnev) [reply default.nix]: What is difference between `nativeBuildIntputs` vs `buildInputs`?

Comment (kirelagin) [reply default.nix]: It becomes relevant if you thing about cross-compilation, when you use one computer to generate code that will run on a different computer. `buildInputs` are things that will “run” on the target computer, i.e. all kinds of runtime dependencies, such as libraries used. `nativeBuildInputs` are things that will run on the build computer, most notably build tools such as the compiler, etc.

**Pull request merged**